### PR TITLE
Python library learns how to use virtual chunk containers

### PR DESCRIPTION
**Pull request by paraseba**

### CODE REVIEW

File: icechunk-python/python/icechunk/_icechunk_python.pyi [hunk 1]
```diff
@@ -3,6 +3,44 @@ import datetime
 from collections.abc import AsyncGenerator
 from enum import Enum
 
+class S3CompatibleOptions:
```
Comment (paraseba) [inline icechunk-python/python/icechunk/_icechunk_python.pyi:6]: this code was moved within the file

File: icechunk-python/python/icechunk/_icechunk_python.pyi [hunk 2]
```diff
@@ -3,6 +3,44 @@ import datetime
...
+    class Gcs:
+        def __init__(self) -> None: ...
+
+    class Azure:
+        def __init__(self) -> None: ...
+
+    class Tigris:
+        def __init__(self) -> None: ...
+
+class VirtualChunkContainer:
```
Comment (paraseba) [inline icechunk-python/python/icechunk/_icechunk_python.pyi:37]: this is new code

File: icechunk-python/python/icechunk/_icechunk_python.pyi [hunk 3]
```diff
@@ -22,27 +60,39 @@ class RepositoryConfig:
         """
         ...
 
+    @property
```
Comment (paraseba) [inline icechunk-python/python/icechunk/_icechunk_python.pyi:63]: we were missing these

File: icechunk-python/python/icechunk/_icechunk_python.pyi [hunk 4]
```diff
@@ -22,27 +60,39 @@ class RepositoryConfig:
...
+    def clear_virtual_chunk_containers(self) -> None: ...
+
 class PyRepository:
     @classmethod
     def create(
         cls,
         storage: Storage,
         *,
         config: RepositoryConfig | None = None,
+        virtual_chunk_containers: dict[str, Credentials] | None = None,
```
Comment (mpiannucci) [inline icechunk-python/python/icechunk/_icechunk_python.pyi]: should these be named `virtual_chunk_containers` or `virtual_chunk_credentials`?

**Review by mpiannucci — APPROVED** [review]
Looks good! One small naming issue

File: icechunk-python/python/icechunk/_icechunk_python.pyi (hunk 4 above)
Comment (paraseba) [reply icechunk-python/python/icechunk/_icechunk_python.pyi]: good catch ... fixed

**Pull request merged**